Library/Homebrew/rubocops/cask/array_alphabetization.rb

```diff
@@ -0,0 +1,69 @@
+# typed: true
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Cask
+      class ArrayAlphabetization < Base
+        extend AutoCorrector
+
+        def on_send(node)
+          return unless [:zap, :uninstall].include?(name = node.method_name)
+
+          node.each_descendant(:pair).each do |pair|
+            symbols = pair.children.select(&:sym_type?).map(&:value)
+            # For `zap`s, we only care about `trash` arrays.
+            next if name == :zap && !symbols.include?(:trash)
+            # Don't order `uninstall` arrays that contain commands.
+            next if name == :uninstall && symbols.intersect?([:signal, :script, :early_script, :args, :input])
+
+            pair.each_descendant(:array).each do |array|
+              if array.children.length == 1
+                add_offense(array, message: "Avoid single-element arrays by removing the []") do |corrector|
+                  corrector.replace(array.source_range, array.children.first.source)
+                end
+              end
+
+              next if array.children.length <= 1
+
+              sorted_array = sort_array(array.source.split("\n")).join("\n")
+
+              next if array.source == sorted_array
+
+              add_offense(array, message: "The array elements should be ordered alphabetically") do |corrector|
+                corrector.replace(array.source_range, sorted_array)
+              end
+            end
+          end
+        end
+
+        def sort_array(source)
+          # Combine each comment with the line below it so that they remain connected to the line they comment
+          combined_source = source.each_with_index.map do |line, index|
+            if line.strip.start_with?("#") && index < source.length - 1
+              "#{line}\n#{source[index + 1]}"
+            elsif source[index - 1]&.strip&.start_with?("#")
+              nil
+            else
+              line
+            end
+          end.compact
```

Comment:
```suggestion

        def recursively_find_comments(source, index, line)
          if source[index - 1].strip.start_with?("#")
            return recursively_find_comments(source, index - 1, "#{source[index - 1]}\n#{line}")
          end
          return line
        end

        def sort_array(source)
          # Combine each comment with the line(s) below it so that they remain connected to the line they comment
          combined_source = source.each_with_index.map do |line, index|
            if line.strip.start_with?("#")
              next
            end
            next recursively_find_comments(source, index, line)
          end.compact
```

Reply:
Should probably add a test for this case too.

Reply:
Done in [458844a](https://github.com/Homebrew/brew/pull/16365/commits/458844af44d5a85488629b398bca161cd292d574), with a test.